server.go

```diff
@@ -142,12 +142,16 @@ func (s *Server) Finish(ke3 *message.KE3) error {
 	return nil
 }
 
-// SessionKey returns the session key if the previous calls to Init() and Finish() were
-// successful.
+// SessionKey returns the session key if the previous call to Init() was successful.
 func (s *Server) SessionKey() []byte {
 	return s.Ake.SessionKey()
 }
 
+// ExpectedMAC returns the expected client MAC if the previous call to Init() was successful.
+func (s *Server) ExpectedMAC() []byte {
+	return s.Ake.ExpectedMAC()
+}
```

Comment:
It turns out, for my use case, it's useful to have an accessor for the expected MAC. Without it, I would first need to fetch the server's serialized state from a database, rehydrate it using `DeserializeAKEState`, call `Finish`, and then make another call to the database to persist an authenticated session.

I do already have `SessionKey`, so I could theoretically determine the expected MAC by trimming off `SessionKey` as a suffix from the output of `SerializeState`; what's left is the MAC. But that's breaking the abstraction.

By adding this accessor, I can make a single call to the database (in a transaction) to assert that the MAC I have from the client matches the expected MAC, and persist an authenticated session.